### PR TITLE
Add partitioned queues

### PR DESCRIPTION
**Pull request by davidffrench**
### Type of change
- Enhancement / new feature

### Description
Adding new example plans for partitioned/sharded queues. These are for the small, medium and large queues with updated description and partitions of 3. 

image

### Checklist
- [ ] Update/write design documentation in `./documentation/design`
- [ ] Write tests and make sure they pass
- [ ] Update documentation
- [x] Check RBAC rights for Kubernetes / OpenShift roles
- [x] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [x] Update CHANGELOG.md

### CODE REVIEW

File: templates/example-plans/020-StandardInfraConfig-default-with-mqtt.yaml [hunk 1]
```diff
@@ -16,7 +16,7 @@ spec:
       storage: 2Gi
     addressFullPolicy: FAIL
   router:
-    minReplicas: 2
+    minReplicas: 3
```
Comment (lulf) [inline templates/example-plans/020-StandardInfraConfig-default-with-mqtt.yaml]: This seems like an unrelated change. The minReplicas of routers does not affect queue partitions. The partitions setting will only scale up the broker, which is handled by the operator.

File: templates/example-plans/020-StandardInfraConfig-default.yaml [hunk 2]
```diff
@@ -14,7 +14,7 @@ spec:
       storage: 2Gi
     addressFullPolicy: FAIL
   router:
-    minReplicas: 2
+    minReplicas: 3
```
Comment (lulf) [inline templates/example-plans/020-StandardInfraConfig-default.yaml]: Same as above.

File: templates/example-plans/020-StandardInfraConfig-default-with-mqtt.yaml (hunk 1 above)
Comment (davidffrench) [reply templates/example-plans/020-StandardInfraConfig-default-with-mqtt.yaml]: It is, was easier to include it in this PR but should really be a separate PR. would you like me to split it out?

File: templates/example-plans/020-AddressPlan-standard-large-partitioned-queue.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+kind: AddressPlan
+metadata:
+  name: standard-large-partitioned-queue
+  labels:
+    app: enmasse
+spec:
+  displayName: Large HA Queue
+  displayOrder: 6
+  shortDescription: Creates a large HA queue backed by a dedicated broker.
+  longDescription: Creates a large HA queue backed by a dedicated broker. The queue is sharded accross multiple brokers for HA and improved performance. A sharded queue no longer guarantees message ordering.
```
Comment (rgodfrey) [inline templates/example-plans/020-AddressPlan-standard-large-partitioned-queue.yaml]: This "backed by a dedicated broker" isn't true, is it... If the total size is 1.0, but it uses 3 partitions, it will only be using 0.3333 of a broker 3 times

File: templates/example-plans/020-StandardInfraConfig-default-with-mqtt.yaml (hunk 1 above)
Comment (rgodfrey) [reply templates/example-plans/020-StandardInfraConfig-default-with-mqtt.yaml]: what's the rationale for having a different config with a minimum router replica of three anyway... why not 4, 5, 17... etc

Comment (lulf) [reply templates/example-plans/020-StandardInfraConfig-default-with-mqtt.yaml]: I don’t see the point in increasing minreplicas beyond 2 @davidffrench

Comment (davidffrench) [reply templates/example-plans/020-StandardInfraConfig-default-with-mqtt.yaml]: These are good points, 2 routers are sufficient for HA in these sample plans. I have removed this commit.

File: templates/example-plans/020-AddressPlan-standard-large-partitioned-queue.yaml (hunk 3 above)
Comment (davidffrench) [reply templates/example-plans/020-AddressPlan-standard-large-partitioned-queue.yaml]: Good spotting @rgodfrey . I have updated the text now to 
```  
shortDescription: Creates a large HA queue sharing underlying brokers with other queues.
longDescription: Creates a large HA queue sharing underlying brokers with other queues. The queue is sharded accross multiple brokers for HA and improved performance. A sharded queue no longer guarantees
```

**Comment by lulf** [comment]
@enmasse-ci run tests profile=smoke

**Comment by davidffrench** [comment]
Thank you again for the quick review @lulf
